Bottom: ee3eb95074ddf5e284104775524da98213fe495b
Top:    3c7997cdf795cc5c1e9e4299a6a6dc1b4eeacde7
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-10-15 17:35:37 -0700

fix


---
diff --git a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
index aa5bf87e59d..df9c98996a4 100644
--- a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
+++ b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
@@ -683,9 +683,13 @@ Expected<PythonObject>
 PythonDictionary::GetItem(const PythonObject &key) const {
   if (!IsValid())
     return nullDeref();
+#if PY_MAJOR_VERSION >= 3
   PyObject *o = PyDict_GetItemWithError(m_py_obj, key.get());
   if (PyErr_Occurred())
     return exception();
+#else
+  PyObject *o = PyDict_GetItem(m_py_obj, key.get());
+#endif
   if (!o)
     return keyError();
   return Retain<PythonObject>(o);
